src/server/middlewares/tierAccessMiddleware/resolvers.js

```diff
@@ -0,0 +1,222 @@
+import _ from 'lodash';
+import assert from 'assert';
+import { ApolloError, UserInputError } from 'apollo-server';
+import log from '../../logger';
+import config from '../../config';
+import esInstance from '../../es/index';
+import CodedError from '../../utils/error';
+import { isWhitelisted, addTwoFilters } from '../../utils/utils';
+
+const ENCRYPT_COUNT = -1;
+
+const resolverWithAccessibleFilterApplied = (
+  resolve, root, args, context, info, authHelper, filter,
+) => {
+  const appliedFilter = authHelper.applyAccessibleFilter(filter);
+  const newArgs = {
+    ...args,
+    filter: appliedFilter,
+    needEncryptAgg: false,
+  };
+  return resolve(root, newArgs, context, info);
+};
+
+const resolverWithUnaccessibleFilterApplied = (
+  resolve, root, args, context, info, authHelper, filter,
+) => {
+  const appliedFilter = authHelper.applyUnaccessibleFilter(filter);
+  const newArgs = {
+    ...args,
+    filter: appliedFilter,
+    needEncryptAgg: true,
+  };
+  return resolve(root, newArgs, context, info);
+};
+
+export const tierAccessResolver = (
+  {
+    isRawDataQuery,
+    esType,
+    esIndex,
+  },
+) => async (resolve, root, args, context, info) => {
+  try {
+    // Assert that either this index is "regular" access or
+    // that the index has no setting and site-wide config is "regular".
+    const indexConfig = esInstance.getESIndexConfigByName(esIndex);
+    const indexIsRegularAccess = indexConfig.tier_access_level === 'regular';
+    const tierAccessRegularAndNotIndexScoped = !indexConfig.tier_access_level && config.tierAccessLevel === 'regular';
```

Comment:
I think this should be good if just check for `config.tierAccessLevel === 'regular'`?

In `server/config.js` we will delete `config.tierAccessLevel` if all indices has its own `tier_access_level`, if not, we are still honoring the overall `config.tierAccessLevel`